contracts/core/Dispatcher.sol

```diff
@@ -390,17 +411,19 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
         Channel memory channel = portChannelMap[address(receiver)][packet.src.channelId];
 
         if (channel.ordering == ChannelOrder.ORDERED) {
-            if (packet.sequence != nextSequenceAck[address(receiver)][packet.src.channelId]) {
+            if (packet.sequence != nextSequence[address(receiver)][packet.src.channelId].ack) {
                 revert unexpectedPacketSequence();
             }
 
-            nextSequenceAck[address(receiver)][packet.src.channelId] = packet.sequence + 1;
+            unchecked {
+                nextSequence[address(receiver)][packet.src.channelId].ack = packet.sequence + 1;
```

Comment:
Since the `packet` is sent directly through calldata, could there be problems with some inputs? If there's a packet with the sequence set to `max(uint64)`, this would result in an overflow. 

I'm leaning towards suggesting to leave this as checked math for now since I'm not sure what the implications are for an overflow happening here

Reply:
I assumed that this check: `(packet.sequence != nextSequence[address(receiver)][packet.src.channelId].ack)` bounded it.

Reply:
Ah ok, I think this is valid